ambry-quota/src/main/java/com/github/ambry/quota/QuotaUtils.java

```diff
@@ -90,6 +90,15 @@ public static QuotaChargeCallback buildQuotaChargeCallback(RestRequest restReque
     }
   }
 
+  /**
+   * @param quotaChargeCallback {@link QuotaChargeCallback} object.
+   * @return {@code true} if the charge needs to happen AFTER a request is processed. {@code false} otherwise.
+   */
+  public static boolean postProcessCharge(QuotaChargeCallback quotaChargeCallback) {
+    boolean retval = quotaChargeCallback != null && !quotaChargeCallback.getQuotaConfig().bandwidthThrottlingFeatureEnabled;
```

Comment:
Why do we return true when the bandwidthThrottlingFeature is not enabled?

Do we need retval here?

Reply:
So there are going to be two implementations of QuotaChargeCallback going forward.
One will be what we have currently. It charges for quota, after the chunks are processed by the server. So it charged "post process". Its named `RejectingQuotaChargeCallback`, which is a bad name actually. 

The new "bandwidth throttling" based implementation will charge for quota "before" sending the chunk request to server. But note that the new implementation will also reject requests, because for storage quota we will reject the requests if they exceed quota.

So in summary the right set of changes are
1. Rename `RejectingQuotachargeCallback` to `PostProcessQuotaChargeCallback`.
2. New implementation should be called `PreProcessQuotaChargeCallback`.
3. If bandwidth throttling is enabled, then don't do stuff related to `PostProcessQuotaChargeCallback`. Or in other words, if bandwidth throttling is not enabled, then do the `PostProcessQuotaChargeCallback` stuff.

The logic above implements the point 3. PR [2069](https://github.com/linkedin/ambry/pull/2069/files) implements the point 1. PR [2070](https://github.com/linkedin/ambry/pull/2069/files) implements the point 2.